File: src/main/java/no/nav/fo/veilarbregistrering/domain/SykmeldtRegistrering.java
package no.nav.fo.veilarbregistrering.domain;

import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.ToString;
import lombok.experimental.Accessors;
import no.nav.fo.veilarbregistrering.domain.besvarelse.Besvarelse;
import no.nav.fo.veilarbregistrering.domain.besvarelse.Stilling;

import java.time.LocalDateTime;
import java.util.List;

import static no.nav.fo.veilarbregistrering.domain.BrukerRegistreringType.SYKMELDT;

@Data
@Accessors(chain = true)
@ToString
@EqualsAndHashCode
public class SykmeldtRegistrering extends BrukerRegistrering {
    long id;
    LocalDateTime opprettetDato;
    Besvarelse besvarelse;
    List<TekstForSporsmal> teksterForBesvarelse;

    @Override
    BrukerRegistreringType hentType() {
        return SYKMELDT;
    }
}

File: src/main/kotlin/no/nav/fo/veilarbregistrering/registrering/sykmeldt/SykmeldtRegistrering.kt
package no.nav.fo.veilarbregistrering.registrering.sykmeldt

import no.nav.fo.veilarbregistrering.besvarelse.Besvarelse
import no.nav.fo.veilarbregistrering.registrering.BrukerRegistreringType
import no.nav.fo.veilarbregistrering.registrering.bruker.BrukerRegistrering
import no.nav.fo.veilarbregistrering.registrering.bruker.TekstForSporsmal
import no.nav.fo.veilarbregistrering.registrering.manuell.Veileder
import java.time.LocalDateTime

data class SykmeldtRegistrering(
    override val id: Long = 0,
    val opprettetDato: LocalDateTime = LocalDateTime.now(),
    val besvarelse: Besvarelse,
    val teksterForBesvarelse: List<TekstForSporsmal>,
    override var manueltRegistrertAv: Veileder? = null,
) : BrukerRegistrering() {
    override fun hentType(): BrukerRegistreringType {
        return BrukerRegistreringType.SYKMELDT
    }

    override fun toString(): String {
        return "SykmeldtRegistrering(id=" + id + ", opprettetDato=" + opprettetDato + ", besvarelse=" + besvarelse + ", teksterForBesvarelse=" + teksterForBesvarelse + ")"
    }
}